### PR TITLE
 Don't drop function argument debug info in the LoadableByAddress transformation

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://problem/36392957

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test and merge

File: test/DebugInfo/LoadableByAddress-argument.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  let field2 : Int64 = 2
+  let field3 : Int64 = 3
+  let field4 : Int64 = 4
+  let field5 : Int64 = 5
+  let field6 : Int64 = 6
+  let field7 : Int64 = 7
+  let field8 : Int64 = 8
+}
+
+// CHECK: !DILocalVariable(name: "largeArg", arg: 1
```
Comment (dcci) [inline test/DebugInfo/LoadableByAddress-argument.swift]: As this is caused by `LoadableByAddress`, can we also dump the relevant SIL output after the transformations? (not the whole function).

**Review by dcci — APPROVED** [review]
The fix looks fine but I have a small comment inline.

File: test/DebugInfo/LoadableByAddress-argument.swift (hunk 1 above)
Comment (adrian-prantl) [reply test/DebugInfo/LoadableByAddress-argument.swift]: That would be nice, but there is an outstanding bug (rdar://problem/35800325) that we can't dump the output of SIL transformations that happen in IRGen.

**Comment by milseman** [comment]
I'm very happy to see this fix. As far as reviewing the patch is concerned, @shajrawi is better.

**Review by shajrawi — COMMENTED** [review]
LGTM!

File: test/DebugInfo/LoadableByAddress-argument.swift (hunk 1 above)
Comment (dcci) [reply test/DebugInfo/LoadableByAddress-argument.swift]: OK, cool, thanks!

**Pull request merged**